src/platform/Darwin/BleConnectionDelegateImpl.mm

```diff
@@ -460,7 +483,10 @@ - (void)disconnect
         return;
     }
 
-    _mBleLayer->CloseAllBleConnections();
+    dispatch_sync(_chipWorkQueue, ^{
```

Comment:
Hmm.  When does this get called?  Are we guaranteed the chip work queue is running at this point?

Reply:
This is normally called when:
 1. someone tries to do a new connection but there was a previous connection attempt. It is called from the ble work queue (it used to be called from the chip work queue before this PR).
 2. the target device to connect to is not found. It is called from the ble work queue
 3. the ble physical interface is turned off. It is called from the ble work queue

fwiw I have rewritten this particular part to do an async dispatch to the chip work queue.